test/accumulators_fraction_test.cpp

```diff
@@ -19,15 +19,25 @@ void run_tests() {
 
   // ctor
   {
-    f_t f(1, 2);
-    BOOST_TEST_EQ(f.successes(), 1);
-    BOOST_TEST_EQ(f.failures(), 2);
+    f_t f1(3, 1);
+    f_t f2(3.0, 5.0);
+    f_t f3(1.0, 3.0);
+    BOOST_TEST_EQ(f1.successes(), 3); BOOST_TEST_EQ(f1.failures(), 1); BOOST_TEST_EQ(f1.successes(), 3.0); BOOST_TEST_EQ(f1.failures(), 1.0);
+    BOOST_TEST_EQ(f2.successes(), 3); BOOST_TEST_EQ(f2.failures(), 5); BOOST_TEST_EQ(f2.successes(), 3.0); BOOST_TEST_EQ(f2.failures(), 5.0);
+    BOOST_TEST_EQ(f3.successes(), 1); BOOST_TEST_EQ(f3.failures(), 3); BOOST_TEST_EQ(f3.successes(), 1.0); BOOST_TEST_EQ(f3.failures(), 3.0);
+    BOOST_TEST_EQ(f1.value(), 0.75);
+    BOOST_TEST_EQ(f2.value(), 0.375);
+    BOOST_TEST_EQ(f3.value(), 0.25);
+    BOOST_TEST_EQ(f1.variance(), 12); BOOST_TEST_EQ(f1.variance(), 12.0);
+    BOOST_TEST_EQ(f2.variance(), 120); BOOST_TEST_EQ(f2.variance(), 120.0);
+    BOOST_TEST_EQ(f3.variance(), 12); BOOST_TEST_EQ(f3.variance(), 12.0);
+    // BOOST_TEST_EQ(f1.confidence_interval().first, 0.5); BOOST_TEST_EQ(f1.confidence_interval().second, 0.9);
   }
 }
 
 int main() {
 
-  run_tests<int>();
+  // run_tests<int>(); // confidence_interval() throws error due to static_assert in binomial_proportion_interval()
```

Comment:
This test exposes an error in the implementation in `fraction`. If `value_type` is an integer, which is perfectly valid, then the confidence interval computer should still return an interval in `double`. To get this working correctly requires a bit of metaprogramming that we can resolve together tomorrow in our call.